packages/contracts/src/contracts/anchor.ts

```diff
@@ -48,8 +54,9 @@ export class AnchorContract {
     const overrides = {
       gasLimit: 6000000,
       gasPrice: utils.toWei('1', 'gwei'),
-      value: '0x16345785D8A0000',
+      value: await this.denomination,
```

Comment:
Maybe I don't know much about getters but why are we awaiting this?

Reply:
This makes a call to the RPC to read the denomination from the contract deployed on the evm. We need to wait for the RPC response.

Reply:
Awesome, so this is mostly to generalise over Substrate chains and EVM, since we get denomination info often asynchronously. Good stuff!